subsys/testsuite/ztest/include/zephyr/ztest_assert.h

```diff
@@ -187,6 +225,39 @@ static inline bool z_zassume(bool cond, const char *default_msg, const char *fil
 #define zassume(cond, default_msg, ...)                                                            \
 	_zassume_va(cond, default_msg, COND_CODE_1(__VA_OPT__(1), (__VA_ARGS__), (NULL)))
 
+/**
+ * @brief If @a cond is false, fail the test but continue its execution.
+ *
+ * You probably don't need to call this macro directly. You should
+ * instead use zexpect_{condition} macros below.
+ *
+ * Note that when CONFIG_MULTITHREADING=n macro returns from the function. It's presupposed that
```

Comment:
I see that you copied this from `_zassert_base` but changed "expected" to "presupposed" to avoid confusion.
1. Nit: With the addition of this API, the previous comments may also be confusing and should be changed.
2. Nit: IMO, this whole paragraph (and its analogs) could be more clearly worded as something like, "With `CONFIG_MULTITHREADING=n`, this macro returns from the enclosing function and should only be used within a test function."
3. Not a nit: If `_zexpect_base` can return from the enclosing function, is "expect" any different than "assert"? Should this even build with `CONFIG_MULTITHREADING=n`?

Reply:
Good point. `CONFIG_MULTITHREADING` should have no impact on this function. Removing that part of the docstring since it always returns.